java/README.md

```diff
@@ -0,0 +1,201 @@
+# Velox4J: Java Bindings for Velox
+
+![badge](https://github.com/facebookincubator/velox/actions/workflows/java-ut-java.yml/badge.svg?branch=main)
+![badge](https://github.com/facebookincubator/velox/actions/workflows/java-ut-cpp.yml/badge.svg?branch=main)
+
+## Introduction
+
+### What is Velox4J?
+
+Velox4J is the Java bindings for Velox. It enables JVM applications to directly invoke Velox's
+functionalities without writing and maintaining any C++ / JNI code.
+
+## Design
+
+Velox4J is designed within the following manners:
+
+### Portable
+
+Velox4J is designed to be portable. The eventual goal is to make one Velox4J release to be
+shipped onto difference platforms without rebuilding the Jar file.
+
+### Seamless Velox API Mapping
+
+Velox4J directly adopts Velox's existing JSON serde framework and implements the following
+JSON-serializable Velox components in Java-side:
+
+- Data types
+- Query plans
+- Expressions
+- Connectors
+
+With the help of Velox's own JSON serde, there will be no re-interpreting layer for query plans
+in Velox4J's C++ code base. Which means, the Java side Velox components defined in Velox4J's
+Java code will be 1-on-1 mapped to Velox's associated components. The design makes Velox4J's
+code base even small, and any new Velox features easy to add to Velox4J.
+
+### Compatible With Arrow Java
+
+Velox4J is compatible with [Apache Arrow's Java implementation](https://arrow.apache.org/java/). Built-in utilities converting between
```

Comment:
As discussed [here](https://github.com/facebookincubator/velox/discussions/12808#discussioncomment-12650164), introducing the Arrow data format can be costly. Could we consider adopting Gluten's implementation, which directly converts the Spark format to the Velox format, thereby bypassing the Arrow format? This approach might require applications to implement the format conversion themselves. However, it could lead to performance improvements. Alternatively, we could offer an API that allows users to implement the data conversion without directly using the Arrow format.

Reply:
> Could we consider adopting Gluten's implementation, which directly converts the Spark format to the Velox format, thereby bypassing the Arrow format?

This is feasible. These native Spark2Velox / Flink2Velox implementations can be added directly to Java bindings code based but with some extensible API designs. For Spark, Velox already provides the transition APIs so all we need is to expose them from C++ to Java.